### ISSUE TITLE
Unhandled Exception: RPCError: got code -32602 with msg "too many arguments, want at most 1"

### ISSUE THREAD

**Issue by YDJSZY** [simolus3/web3dart#61]
I have upgraded the libary. This is my code:
 var gas = await web3client.estimateGas();
It throw error "got code -32602 with msg "too many arguments, want at most 1".
This is detail error:
ui_dart_state.cc(148)] Unhandled Exception: RPCError: got code -32602 with msg "too many arguments, want at most 1".
#0      JsonRPC.call (package:web3dart/json_rpc.dart:49:7)
<asynchronous suspension>
#1      Web3Client._makeRPCCall (package:web3dart/src/core/client.dart:58:35)
<asynchronous suspension>
#2      Web3Client.estimateGas (package:web3dart/src/core/client.dart:325:29)

**Comment by iceman750** [comment]
Check this code out:
https://github.com/simolus3/web3dart/commit/82151789b64b1829d75b1b72c6688a7e1861a712

**Comment by YDJSZY** [comment]
@iceman750 What do you mean? Now my web3dart version is 1.2.0

**Comment by simolus3** [comment]
Which Ethereum client are you connecting to? The rpc docs specify two parameters on the `eth_estimate` call, so I'm wondering why a node would reject that

**Comment by YDJSZY** [comment]
@simolus3 I connecting to test network.

**Comment by YDJSZY** [comment]
I found this code in web3.js:
`new Method({
            name: 'estimateGas',
            call: 'eth_estimateGas',
            params: 1,
            inputFormatter: [formatter.inputCallFormatter],
            outputFormatter: utils.hexToNumber
        })`.

**Comment by YDJSZY** [comment]
I think that rpc specify one parameters. I found it in [https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_estimategas](url)

**Comment by simolus3** [comment]
The docs from `eth_estimateGas` say that they support the same parameters as [`eth_call`](https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_call), which has a second argument to specify the block number, so I assumed that the block number argument was supported on the estimate call as well.
The node I used to test this didn't complain about the second parameter, but infura does. I've just released `1.2.1` which deprecates the `atBlock` parameter and doesn't send it. This should fix the problem.
